### PR TITLE
Add codecov upload

### PR DESCRIPTION
**Pull request by dagardner-nv**
- Fix bug preventing the coverage report from running.
- Coverage report needs the `.cache` dir from the build step.
- Add `--cap-add=sys_nice` arg to benchmark stage (unrelated fix) noticed that `bench_srf.x` needs it for memory policy
- Remove unneeded `rm -rf` of the `build` and `.cache` dirs on startup, this is no longer needed.

### CODE REVIEW

**Review by mdemoret-nv — CHANGES_REQUESTED** [review]
Couple of things.

File: ci/scripts/github/common.sh [hunk 1]
```diff
@@ -101,6 +101,22 @@ function fetch_s3() {
...
+
+    gpuci_logger "Verifying codecov"
+    VERSION=$(grep -o 'VERSION=\"[0-9\.]*\"' ${CODECOVE_SH} | cut -d'"' -f2);
+
+    pushd ${WORKSPACE_TMP}
+    for i in 1 256 512
+    do
+        shasum -a $i -c --ignore-missing <(curl -s "https://raw.githubusercontent.com/codecov/codecov-bash/${VERSION}/SHA${i}SUM")
+    done
+    popd
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply ci/scripts/github/common.sh:117]: It's checking sha1, sha256 and sha512 
This is what was in their docs on how to verify it:
https://about.codecov.io/blog/validating-the-bash-script-on-ci/?utm_medium=article&utm_source=blog&utm_campaign=security-incident&utm_content=validating-bash-on-ci&utm_term=clicked-inline-cta

File: codecov.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#Configuration File for CodeCov
```
[the remark this answers is not in the document]
Comment (dagardner-nv) [reply codecov.yml]: I moved it to the `.github` dir

**Review by mdemoret-nv — APPROVED** [review]
LGTM

**Comment by mdemoret-nv** [comment]
@gpucibot merge

**Pull request merged**